src/scenic/core/simulators.py

```diff
@@ -537,6 +547,10 @@ def updateObjects(self):
             for prop in self.mutableProperties(obj):
                 values[prop] = getattr(obj, prop)
 
+            # Sanitize some 2D values
+            if isinstance(obj, Object2D):
+                values["position"] = Vector(values["position"].x, values["position"].y, 0)
```

Comment:
Since `_copyWith` runs specifier resolution anyway (maybe something we should think about improving), what about having `Object2D._specify` null the Z coordinate of `position`? Then this module wouldn't need to know about 2D mode.